sentry-android-timber/src/main/java/io/sentry/android/timber/SentryTimberTree.kt

```diff
@@ -17,42 +17,218 @@ class SentryTimberTree(
+        vararg args: Any?
+    ) {
+        logWithSentry(Log.ERROR, null, message, *args)
+    }
+
+    /** Log an error exception and a message with optional format args. */
+    override fun e(
+        t: Throwable?,
+        message: String?,
+        vararg args: Any?
+    ) {
+        logWithSentry(Log.ERROR, t, message, *args)
+    }
+
+    /** Log an error exception. */
+    override fun e(t: Throwable?) {
+        logWithSentry(Log.ERROR, t, null)
+    }
+
+    /** Log an assert message with optional format args. */
+    override fun wtf(
+        message: String?,
+        vararg args: Any?
+    ) {
+        logWithSentry(Log.ASSERT, null, message, *args)
+    }
+
+    /** Log an assert exception and a message with optional format args. */
+    override fun wtf(
+        t: Throwable?,
+        message: String?,
+        vararg args: Any?
+    ) {
+        logWithSentry(Log.ASSERT, t, message, *args)
+    }
+
+    /** Log an assert exception. */
+    override fun wtf(t: Throwable?) {
+        logWithSentry(Log.ASSERT, t, null)
+    }
+
+    /** Log at `priority` a message with optional format args. */
+    override fun log(
+        priority: Int,
+        message: String?,
+        vararg args: Any?
+    ) {
+        logWithSentry(priority, null, message, *args)
+    }
+
+    /** Log at `priority` an exception and a message with optional format args. */
+    override fun log(
+        priority: Int,
+        t: Throwable?,
+        message: String?,
+        vararg args: Any?
+    ) {
+        logWithSentry(priority, t, message, *args)
+    }
+
+    /** Log at `priority` an exception. */
+    override fun log(
+        priority: Int,
+        t: Throwable?
+    ) {
+        logWithSentry(priority, t, null)
+    }
+
+    override fun log(
+        priority: Int,
+        tag: String?,
+        message: String,
+        t: Throwable?
+    ) {
+        // no-op as we've overridden all the methods
+    }
+
+    private fun logWithSentry(
+        priority: Int,
+        throwable: Throwable?,
+        message: String?,
+        vararg args: Any?
+    ) {
+        if (message.isNullOrEmpty() && throwable == null) {
+            return // Swallow message if it's null and there's no throwable
+        }
 
-    /**
-     * Captures a Sentry Event if the min. level is equal or higher than the min. required level.
-     */
-    override fun log(priority: Int, tag: String?, message: String, throwable: Throwable?) {
         val level = getSentryLevel(priority)
+        val sentryMessage = Message().apply {
+            this.message = message
+            if (message != null && args.isNotEmpty()) {
+                this.formatted = message.format(*args)
+            }
+            this.params = args.mapNotNull { it.toString() }
+        }
 
-        captureEvent(level, tag, message, throwable)
-        addBreadcrumb(level, message)
+        addBreadcrumb(level, sentryMessage)
+        captureEvent(level, sentryMessage, throwable)
```

Comment:
The order here matters, first you capture the event and then add the breadcrumb, otherwise, the crumb is redundant.

Reply:
hm, actually from my tests, the breadcrumb was captured for the next event, is this how it's designed then?

Reply:
1. event
you capture the error/message (no breadcrumb about the error/message itself).
then you add the breadcrumb.

2. event
you capture the error/message (this time, there's a breadcrumb about event 1.)

This should be the way to go.

Reply:
Alright, clear. A follow-up question then: if the event was logged without a message, but just with a throwable (e.g. `Timber.e(Throwable("Test"))`), should we add the breadcrumb or not? 

Because the message is null in this case, so we either do not add it, or we put a stacktrace (or part of it) into the breadcrumb message.

Reply:
If there's a message, let's use the message, otherwise let's create a message out of the throwable, but not with the stack trace, only with the description of the error, and then we change the type and category of this crumb, following the `error` type here https://develop.sentry.dev/sdk/event-payloads/breadcrumbs/

Reply:
+1 on what Manoel said here ^ 🚀

Reply:
Changed it, had to also additionally set category to `exception`, to align with when an error event happens (otherwise it'd show `generic` category as on the SS below)
![image](https://user-images.githubusercontent.com/4999776/152758539-b214441b-5936-4b17-bd4f-c3bf27bcc6fe.png)